integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -440,14 +448,22 @@ object SecondaryIndexUtil {
         val loadFolderDetailsArray = SegmentStatusManager.readLoadMetadata(indexTable
```

Comment:
reading should be inside of locking

Reply:
Agree. Modified.

Reply:
Also handled similar issue in SecondaryIndexUtil#triggerCompaction